crates/uv-configuration/src/name_specifiers.rs

```diff
@@ -59,6 +59,22 @@ impl<'de> serde::Deserialize<'de> for PackageNameSpecifier {
     }
 }
 
+#[cfg(feature = "schemars")]
+impl schemars::JsonSchema for PackageNameSpecifier {
+    fn schema_name() -> String {
+        "PackageNameSpecifier".to_string()
+    }
+
+    fn json_schema(_gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        schemars::schema::SchemaObject {
+            instance_type: Some(schemars::schema::InstanceType::String.into()),
+            format: Some("package name or `:all:` or `:none:`".to_string()),
```

Comment:
This should be `pattern: Some(r"^(:none:|:all:|([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9._-]*[a-zA-Z0-9]))$".to_string())` with a reference to https://packaging.python.org/en/latest/specifications/name-normalization/#name-format